model-engine/model_engine_server/infra/gateways/s3_llm_artifact_gateway.py

```diff
@@ -17,6 +18,21 @@ def _get_s3_resource(self, kwargs):
         resource = session.resource("s3")
         return resource
 
+    def list_files(self, path: str, **kwargs) -> List[str]:
+        s3 = self._get_s3_resource(kwargs)
+        parsed_remote = parse_attachment_url(path)
+        bucket = parsed_remote.bucket
+        key = parsed_remote.key
+        try:
+            # From here: https://dev.to/aws-builders/how-to-list-contents-of-s3-bucket-using-boto3-python-47mm
+            files = [
+                bucket_object["Key"]
+                for bucket_object in s3.list_objects_v2(Bucket=bucket, Prefix=key)["Contents"]
+            ]
+        except Exception as e:  # type: ignore
+            raise e
```

Comment:
This doesn't seem to do anything? We're just catching the exception and raising it again?

Reply:
😅 taken out